app/views/helpers/summary/Card7ForTransport.scala

```diff
@@ -0,0 +1,241 @@
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class Card7ForTransport @Inject() (govukSummaryList: GovukSummaryList) extends SummaryHelper {
+
+  def eval(declaration: ExportsDeclaration, actionsEnabled: Boolean = true)(implicit messages: Messages): Html = {
+    val transport = declaration.transport
+
+    val hasData = transport.expressConsignment.isDefined ||
+      inlandModeOfTransportCode(declaration).isDefined ||
+      declaration.locations.warehouseIdentification.isDefined ||
+      declaration.locations.supervisingCustomsOffice.isDefined ||
+      declaration.locations.inlandOrBorder.isDefined ||
+      declaration.locations.inlandModeOfTransportCode.isDefined ||
+      transport.meansOfTransportOnDepartureType.isDefined ||
+      transport.meansOfTransportOnDepartureIDNumber.isDefined ||
+      transport.meansOfTransportCrossingTheBorderType.isDefined ||
+      transport.meansOfTransportCrossingTheBorderIDNumber.isDefined ||
+      transport.transportPayment.isDefined ||
+      transport.borderModeOfTransportCode.isDefined ||
+      transport.transportCrossingTheBorderNationality.isDefined ||
+      transport.containers.isDefined
+
+    if (hasData) displayCard(declaration, actionsEnabled) else HtmlFormat.empty
+  }
+
+  private def displayCard(declaration: ExportsDeclaration, actionsEnabled: Boolean)(implicit messages: Messages): Html =
+    govukSummaryList(SummaryList(rows(declaration, actionsEnabled), card("transport")))
+
+  private def rows(declaration: ExportsDeclaration, actionsEnabled: Boolean)(implicit messages: Messages): Seq[SummaryListRow] =
+    (List(
+      borderTransport(declaration.transport, actionsEnabled),
+      warehouseIdentification(declaration, actionsEnabled),
+      supervisingCustomsOffice(declaration, actionsEnabled),
+      inlandOrBorder(declaration, actionsEnabled),
+      inlandModeOfTransport(declaration, actionsEnabled),
+      transportReference(declaration.transport, actionsEnabled),
+      activeTransportType(declaration.transport, actionsEnabled),
+      transportPayment(declaration.transport, actionsEnabled),
+      transportCrossingTheBorder(declaration.transport, actionsEnabled),
+      expressConsignment(declaration.transport, actionsEnabled)
+    ) ++ containers(declaration.transport, actionsEnabled)).flatten
+
+  private def inlandModeOfTransportCode(declaration: ExportsDeclaration): Option[InlandModeOfTransportCode] =
+    declaration.locations.inlandModeOfTransportCode.filterNot(_.inlandModeOfTransportCode contains Empty)
+
+  private def borderTransport(transport: Transport, actionsEnabled: Boolean)(implicit messages: Messages): Option[SummaryListRow] =
+    transport.borderModeOfTransportCode.map { borderModeOfTransportCode =>
+      SummaryListRow(
+        key("transport.departure.transportCode.header"),
+        valueKey(s"declaration.summary.transport.departure.transportCode.${borderModeOfTransportCode.getCodeValue}"),
+        classes = "borderTransport",
```

Comment:
May I ask you to follow the same pattern I'm using? Even to respect the standard way to name CSS classes.
They are normally lower case with words separated by hyphens. So `border-transport`, `warehouse-id`, `supervising-office` and so on.

Reply:
No problem! Accident rather than by design